src/repositories/identityRepository.ts

```diff
@@ -1,14 +1,66 @@
-import { QueryResult, runInsertQuery } from "../lib/mysqlDatabase";
+import { escape } from "mysql";
+import { runSelectQuery } from "../lib/mysqlDatabase";
 import { MySQLRepository } from "../lib/MySQLRepository";
+import { wsProviderRepository } from "../repositories/wsProviderRepository";
+import { chainRepository } from "../repositories/chainRepository";
+import { QueryResult, runInsertQuery } from "../lib/mysqlDatabase";
 import { IdentityEntity } from "../types/entities/IdentityEntity";
 import { accountRepository } from "../repositories/accountRepository";
 import { Identity } from "@npmjs_tdsoftware/subidentity";
+import { AccountEntity } from "../types/entities/AccountEntity";
 
 class IdentityRepository extends MySQLRepository<IdentityEntity> {
     get tableName(): string {
         return "identity";
     }
 
+    async getRowCount(wsProvider: string, searchKey: string): Promise<number> {
+        const query = `SELECT COUNT(*)
+                        FROM ${this.tableName}
+                        INNER JOIN ${accountRepository.tableName} ON ${this.tableName}.account_id = ${accountRepository.tableName}.id 
+                        INNER JOIN ${chainRepository.tableName} ON ${accountRepository.tableName}.chain_id = ${chainRepository.tableName}.id
+                        INNER JOIN ${wsProviderRepository.tableName} ON ${chainRepository.tableName}.id = ${wsProviderRepository.tableName}.chain_id
+                        WHERE ${wsProviderRepository.tableName}.address=${escape(wsProvider)}
+                        AND (${this.tableName}.display LIKE "%${searchKey}%" 
+                            OR ${this.tableName}.legal LIKE "%${searchKey}%"
+                            OR ${this.tableName}.address LIKE "%${searchKey}%"
+                            OR ${this.tableName}.riot LIKE "%${searchKey}%"
+                            OR ${this.tableName}.twitter LIKE "%${searchKey}%"
+                            OR ${this.tableName}.web LIKE "%${searchKey}%"
+                            OR ${this.tableName}.email LIKE "%${searchKey}%")
+                        ORDER BY ${this.tableName}.id`;
+        const queryResult = (await runSelectQuery<number>(query))[0];
+        const resultJson = Object.values(JSON.parse(JSON.stringify(queryResult)));
+        return resultJson[0] as number;
+    }
+
+    async searchByWsProviderAndKey(wsProvider: string, searchKey: string, offset: number, limit: number): Promise<IdentityEntity[]> {
+        const query = `SELECT 
+                            ${chainRepository.tableName}.chain_name,
```

Comment:
you are returning IdentityEntity and chain_name is not part of that interface. Maybe you need to add a dedicated interface for that?